spec/services/datetime_parser_spec.rb

```diff
@@ -92,4 +92,50 @@
       end
     end
   end
+
+  describe "#issues_for" do
+    let(:date_issues) { parser.issues_for(:schedule_date) }
+    let(:time_issues) { parser.issues_for(:schedule_time) }
+
+    context "an invalid time is provided" do
+      let(:parser) { DatetimeParser.new(date: valid_date, time: nil) }
+
+      it "finds issues for schedule_time" do
+        parser.parse
+
+        expect(date_issues.count).to eq(0)
+        expect(time_issues.count).to eq(1)
+        expect(time_issues.first).to eq("Enter a valid time")
```

Comment:
If `issues_for` is returning the first one issue for a type, then you shouldn't need to do `.first` here.

Reply:
It returns an array of issues (as that's what's expected by the frontend components by convention). In practice we typically return a 1-element array.